### ISSUE TITLE
gpu problem while training 

### ISSUE THREAD

**Issue by FAFACHR** [dmlc/gluon-cv#768]
Hello @zhreshold , I'm running the training script of mask-rcnn on my own dataset that contain  2 classes, and I'm using the classifier 'resnet50_v1b-0ecdba34.params' like the tutorial in the documentation of gluoncv but instead of coco dataset I'm using my own dataset. The problem is when I run the training with 1 gpu the output result is kind of weird, this is the result:
![Screenshot from 2019-05-07 13-51-48](https://user-images.githubusercontent.com/34231991/57297208-48986980-70cf-11e9-8a39-96370866bad5.png)

but if I run the same code with multiple gpus the output seems normal, this is the result:
![1](https://user-images.githubusercontent.com/34231991/57297298-8e553200-70cf-11e9-9b5e-1872ba35c56f.png)

Can you explain me why I got these result on 1 gpu? and Thank you in advanced

**Comment by zhreshold** [comment]
batchsize is different

**Comment by FAFACHR** [comment]
Thank you @zhreshold for the reply but how can I set the same batch size in the 2 cases? And how the batch size is taken in the script ? Is it related to the number of gpus?

**Comment by FAFACHR** [comment]
Thank you @zhreshold for the reply, so what's the best number of batch size I need to use to have the best result in the 2 cases ? because I saw in the script that the batch size is the number of gpus used.

**Comment by zhreshold** [comment]
You will want at least batch size 4 in order to train a reasonable model

**Comment by FAFACHR** [comment]
Thank you @zhreshold , I have another question and sorry for asking a lot.
I set the same batch_size for the 2 cases multiple gpus and single gpu so why I got a great result from epochs 0 with 2 gpus and a bad result with 1 gpu?
